### PR TITLE
multi_threaded feature rename

### LINKED ISSUE #12966: multi-threaded feature name is kebab case but all others are snake case
## Bevy version

0.14

## What went wrong

See https://github.com/bevyengine/bevy/blob/62f2a73cac70237c83054345a26b19c9e0a0ee2f/Cargo.toml#L70

### PR DESCRIPTION
**Pull request by andristarr**
# Objective

Fixes #12966

## Solution

Renaming multi_threaded feature to match snake case

## Migration Guide

Bevy feature multi-threaded should be refered to multi_threaded from now on.

### CODE REVIEW

**Comment by andristarr** [comment]
basis-universal and the symphonia-* features seem to be package names so I suspect having kebab case there is fine?

Do we also want to rename the internal name of the packages? Not sure the exact technical term, but do we want `bevy_internal` to export `multi_threaded` or we are content with kebab case there?

**Review by james7132 — CHANGES_REQUESTED** [review]
All of the cfg annotations that rely on the feature should be updated too.

**Comment by mockersf** [comment]
This PR only rename the feature on the bevy crate, so any use is not impacted.

But I would prefer to rename the feature in all crates for consistency.

File: crates/bevy_tasks/README.md [hunk 1]
```diff
@@ -19,17 +19,17 @@ It is based on [`async-executor`][async-executor], a lightweight executor that a
 
 ## Usage
 
-In order to be able to optimize task execution in multi-threaded environments,
+In order to be able to optimize task execution in multi_threaded environments,
```
Comment (james7132) [inline crates/bevy_tasks/README.md:22]: This doesn't need to be changed.

**Review by james7132 — APPROVED** [review]

File: crates/bevy_tasks/README.md [hunk 2]
```diff
@@ -25,11 +25,11 @@ bevy provides three different thread pools via which tasks of different kinds ca
 This currently applies to WASM targets.)
 The determining factor for what kind of work should go in each pool is latency requirements:
 
-* For CPU-intensive work (tasks that generally spin until completion) we have a standard
+- For CPU-intensive work (tasks that generally spin until completion) we have a standard
   [`ComputeTaskPool`] and an [`AsyncComputeTaskPool`]. Work that does not need to be completed to
   present the next frame should go to the [`AsyncComputeTaskPool`].
 
-* For IO-intensive work (tasks that spend very little time in a "woken" state) we have an
+- For IO-intensive work (tasks that spend very little time in a "woken" state) we have an
```
Comment (mockersf) [inline crates/bevy_tasks/README.md:32]: those changes seems unrelated

Comment (andristarr) [reply crates/bevy_tasks/README.md]: This is now addressed

**Comment by andristarr** [comment]
@mockersf this should be ready for re-review

**Review by mockersf — APPROVED** [review]

**Comment by Aronry** [comment]
Hey! I know this is probably not Github etiquette, but nonetheless I felt compelled to comment that these sorts of changes are breaking, and they cause allot of pain and suffering for people.

I know it feels good to get a green dot on the github calendar and a "pull-shark" Github badge or whatever, but this sort of stuff hurts devs and wastes work hours. "multi-threaded" was more correct anyway.

**Pull request merged**